packages/ember/index.js

```diff
@@ -175,9 +176,30 @@ if (EMBER_EXTEND_PROTOTYPES) {
 Ember.getOwner = getOwner;
 Ember.setOwner = setOwner;
 Ember.Application = Application;
-Ember.DefaultResolver = Ember.Resolver = Resolver;
 Ember.ApplicationInstance = ApplicationInstance;
 
+Object.defineProperty(Ember, 'Resolver', {
+  get() {
+    deprecate(
+      'Using the globals resolver is deprecated. Use the ember-resolver package instead. See https://deprecations.emberjs.com/v3.x#toc_ember-deprecate-globals-resolver',
+      false,
+      {
+        id: 'ember.globals-resolver',
+        until: '4.0.0',
+        url: 'https://deprecations.emberjs.com/v3.x#toc_ember-deprecate-globals-resolver',
+      }
+    );
+
+    return Resolver;
+  },
+});
```

Comment:
Since we issue a deprecation in `init`, wouldn't we end up with two deprecations for _roughly_ the same thing?

Reply:
One would be for creating an application without specifying a resolver, the other for accessing through Ember.Resolver or Ember.DefaultResolver. I generally think most users will not likely run into both unless they specifically request Ember.DefaultResolver as their resolver during application.create.

Reply:
Ya, agreed. Thank you!